util/src/main/java/io/zeebe/util/sched/ActorTask.java

```diff
@@ -331,7 +329,7 @@ boolean tryWait() {
     // take copy of subscriptions list: once we set the state to WAITING, the task could be woken up
     // by another
     // thread. That thread could modify the subscriptions array.
-    final ActorSubscription[] subscriptionsCopy = this.subscriptions;
+    final List<ActorSubscription> subscriptionsRef = this.subscriptions;
```

Comment:
I'm not sure about this line. The comment says that we should copy the subscripts to avoid concurrent access. With this change, the subscriptions are not copied any more. Do you need to care about it?

Reply:
Actually this was never a copy. This just stores the reference and this is the same as before, except that the list is not always replaced. But I think it might make sense to just use a thread safe data structure here

Reply:
As discussed I copy the elements on the tryWait. Furthermore I added an check for the thread on add and removing subscriptions